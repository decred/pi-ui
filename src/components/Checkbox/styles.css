.container {
  display: flex;
  justify-content: flex-start;
  position: relative;
  align-items: center;
  cursor: pointer;
  user-select: none;
}

.container input {
  position: absolute;
  opacity: 0;
  cursor: pointer;
  height: 0;
  width: 0;
}

.checkmark {
  margin-right: 0.8rem;
  height: 1.4rem;
  width: 1.4rem;
  border: 1px solid var(--color-gray);
  background-color: transparent;
}

.checkmark:after {
  content: "";
  position: absolute;
  display: none;
}

.container .checkmark:after {
  left: 5px;
  width: 4px;
  height: 9px;
  border: solid var(--color-white);
  border-width: 0 2px 2px 0;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

.container:hover input ~ .checkmark {
  background-color: var(--color-gray-lighter);
}

.container input:checked ~ .checkmark {
  background-color: var(--color-primary);
}

.container input:checked ~ .checkmark:after {
  display: block;
}
