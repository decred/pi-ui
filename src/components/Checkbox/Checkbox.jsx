import { PropTypes } from "prop-types";
import React from "react";
import styles from "./styles.css";

const Checkbox = ({ label, className, checked, style, name, id, ...props }) => (
  <label htmlFor={id} className={styles.container}>
    <input
      type="checkbox"
      checked={checked}
      id={id}
      name={name}
      style={style}
      {...props}
    />
    <span className={styles.checkmark} />
    <span>{label}</span>
  </label>
);

Checkbox.propTypes = {
  label: PropTypes.string,
  className: PropTypes.string,
  style: PropTypes.object,
  checked: PropTypes.bool,
  name: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired
};

export default Checkbox;
