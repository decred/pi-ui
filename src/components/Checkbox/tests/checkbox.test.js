import React from "react";
import Checkbox from "../Checkbox";
import { create } from "react-test-renderer";

describe("Checkbox component", () => {
  test("Matches the snapshot", () => {
    const checkbox = create(<Checkbox label="test" id="test" />);
    expect(checkbox.toJSON()).toMatchSnapshot();
  });
});
