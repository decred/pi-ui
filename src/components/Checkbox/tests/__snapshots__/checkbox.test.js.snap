// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component Matches the snapshot 1`] = `
<label
  className="container"
  htmlFor="test"
>
  <input
    id="test"
    type="checkbox"
  />
  <span
    className="checkmark"
  />
  <span>
    test
  </span>
</label>
`;
